pkg/cmd/admin/groups/sync/rfc2307/ldapinterface.go

```diff
@@ -57,6 +59,9 @@ type LDAPInterface struct {
 	// cachedUsers holds the result of user queries for later reference, indexed on user UID
 	// e.g. this will map an LDAP user UID to the LDAP entry returned from the query made using it
 	cachedUsers map[string]*ldap.Entry
+
+	// Err is used to log errors found interfacing with LDAP
+	Err io.Writer
```

Comment:
Testability, config controlling whether we tolerate a variety of errors (out of scope, not found, etc), the ability to print errors as they happen or accumulate them at the end…